verify/verify.py

```
@@ -431,21 +430,26 @@ async def _log_verify_message(
             if reason:
                 data.add_field(name="Reason", value=reason)
             if log:
-                try:
-                    await log.send(embed=data)
-                except discord.Forbidden:
-                    await log.send(f"**{message}** - {user.id} - {user}")
+                if channel := self._is_valid_channel(log):
+                    try:
+                        await channel.send(embed=data)
+                    except discord.Forbidden:
+                        await channel.send(f"**{message}** - {user.id} - {user}")
 
     async def _verify_user(self, guild: discord.Guild, member: discord.Member):
         """Private method for verifying a user"""
         async with self.config.guild(guild).blocks() as blocked_users:
             if member.id in blocked_users:
                 return False
 
+        log_id = await self.config.guild(guild).logchannel()
         role_id = await self.config.guild(guild).role()
-        role = guild.get_role(role_id)
-        await member.add_roles(role)
-        return True
+        if role := guild.get_role(role_id):
+            await member.add_roles(role)
+            return True
+        elif log_id and (log_channel := self._is_valid_channel(guild.get_channel(log_id))):
```

This shouldn't silently fail - Add logging.